airbyte-integrations/connectors/destination-mssql-v2/src/main/kotlin/io/airbyte/cdk/load/data/sql/AirbyteValueToSqlValue.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.data.sql
+
+import io.airbyte.cdk.load.data.AirbyteValue
+import io.airbyte.cdk.load.data.ArrayValue
+import io.airbyte.cdk.load.data.BooleanValue
+import io.airbyte.cdk.load.data.DateValue
+import io.airbyte.cdk.load.data.IntegerValue
+import io.airbyte.cdk.load.data.NullValue
+import io.airbyte.cdk.load.data.NumberValue
+import io.airbyte.cdk.load.data.ObjectValue
+import io.airbyte.cdk.load.data.StringValue
+import io.airbyte.cdk.load.data.TimeStringUtility.toLocalDate
+import io.airbyte.cdk.load.data.TimeStringUtility.toLocalDateTime
+import io.airbyte.cdk.load.data.TimeStringUtility.toOffset
+import io.airbyte.cdk.load.data.TimeValue
+import io.airbyte.cdk.load.data.TimestampValue
+import io.airbyte.cdk.load.data.UnknownValue
+import java.sql.Date
+import java.sql.Time
+import java.sql.Timestamp
+
+/** CDK pipeline [AirbyteValue] to SQL values converter. */
+class AirbyteValueToSqlValue {
+
+    /**
+     * Converts an [AirbyteValue] to the associated SQL value.
+     *
+     * @param airbyteValue The [AirbyteValue] from an Airbyte record
+     * @return The corresponding SQL value for the given [AirbyteValue].
+     * @throws IllegalArgumentException if the [AirbyteValue] is not supported.
+     */
+    fun convert(airbyteValue: AirbyteValue): Any? {
+        return when (airbyteValue) {
+            is ObjectValue -> {
+                val convertedValues =
+                    airbyteValue.values.entries.associate { (name, value) ->
+                        name to convert(value)
+                    }
+                convertedValues
+            }
+            is ArrayValue -> throw IllegalArgumentException("Array type is not supported")
+            is BooleanValue -> airbyteValue.value
+            is DateValue -> Date.valueOf(toLocalDate(airbyteValue.value))
+            is IntegerValue -> airbyteValue.value
+            is NullValue -> null
+            is NumberValue -> airbyteValue.value.toDouble().toBigDecimal()
+            is StringValue -> airbyteValue.value
+            is TimeValue -> Time.valueOf(toOffset(airbyteValue.value))
+            is TimestampValue -> Timestamp.valueOf(toLocalDateTime(airbyteValue.value))
+            is UnknownValue -> throw IllegalArgumentException("Unknown type is not supported")
```

Comment:
You support the type (LONGVARCHAR) but not the value. UnknownValue contains the original json you can serialize if you want.

Reply:
I don't know what we want to do here.  It looks like the RAW version throws an exception, so I will make the type side match.

Reply:
`.value.serializeToString()` or `.value.serializeToBytes()`

Reply:
I've removed the type support.